src/main/java/it/gov/pagopa/admissibility/service/build/BeneficiaryRule2DroolsRuleImpl.java

```diff
@@ -55,6 +58,23 @@ public DroolsRule apply(Initiative2BuildDTO initiative) {
                     initiative.getBeneficiaryRule().getAutomatedCriteria().stream().map(c -> automatedCriteriaRuleBuild(out.getId(), out.getName(), c)).collect(Collectors.joining("\n\n")))
             );
 
+            InitiativeConfig initiativeConfig = InitiativeConfig.builder()
+                    .initiativeId(initiative.getInitiativeId())
+                    .pdndToken(initiative.getPdndToken())
+                    .automatedCriteriaCodes(getAutomatedCriteriaCodes(initiative.getBeneficiaryRule().getAutomatedCriteria()))
+                    .initiativeBudget(initiative.getGeneral().getBudget())
+                    .beneficiaryInitiativeBudget(initiative.getGeneral().getBeneficiaryBudget())
+                    .status(initiative.getStatus())
+                    .build();
+            if(initiative.getGeneral().getRankingStartDate()==null || initiative.getGeneral().getRankingEndDate()==null) {
+                initiativeConfig.setStartDate(initiative.getGeneral().getStartDate());
+                initiativeConfig.setEndDate(initiative.getGeneral().getEndDate());
+            }else{
+                initiativeConfig.setStartDate(initiative.getGeneral().getRankingStartDate());
+                initiativeConfig.setEndDate(initiative.getGeneral().getRankingEndDate());
+            }
+            out.setInitiativeConfig(initiativeConfig);
```

Comment:
In order to keep clean the code, put the logic to build InitiativeConfig in a separated function